x-pack/plugins/alerting_builtins/server/alert_types/geo_threshold/alert_type.ts

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { schema } from '@kbn/config-schema';
+import { Service } from '../../types';
+import { BUILT_IN_ALERTS_FEATURE_ID } from '../../../common';
+import { getGeoThresholdExecutor } from './geo_threshold';
+import {
+  ActionGroup,
+  AlertServices,
+  ActionVariable,
+  AlertTypeState,
+} from '../../../../alerts/server';
+
+export const GEO_THRESHOLD_ID = '.geo-threshold';
+export type TrackingEvent = 'entered' | 'exited';
+export const ActionGroupId = 'tracking threshold met';
+
+const actionVariableContextToEntityDateTimeLabel = i18n.translate(
+  'xpack.alertingBuiltins.geoThreshold.actionVariableContextToEntityDateTimeLabel',
+  {
+    defaultMessage: `The time the entity was detected in the current boundary`,
```

Comment:
Not sure where these messages appear, but am wondering if they require an ending period.

Reply:
These ultimately populate in a dropdown. To me the period looks strange in that context so I held off on making this change. Happy to reconsider if you feel strongly though!

Reply:
I agree.  The period would look strange in a dropdown.